src/transformers/models/dpt/modeling_dpt.py

```diff
@@ -226,7 +226,9 @@ def _resize_pos_embed(self, posemb, grid_size_height, grid_size_width, start_ind
         posemb_tok = posemb[:, :start_index]
         posemb_grid = posemb[0, start_index:]
 
-        old_grid_size = int(math.sqrt(len(posemb_grid)))
+        def safe_int(x):
+          return x.to(torch.int64) if torch.jit.is_tracing() else int(x)
+        old_grid_size = safe_int(posemb_grid.size(0) ** 0.5)
```

Comment:
It might be necessary to extract this cast function to a utility file, and then re-use across different models. I have a feeling that other instances, like:
![image](https://github.com/huggingface/transformers/assets/26504141/af48ef19-94f1-4189-b8a7-a618c03d5082)

will break execution, since during inference, it may require normal python types